website/project/views/node.py

```diff
@@ -413,6 +419,9 @@ def view_project(auth, node, **kwargs):
     ))
     ret.update(rubeus.collect_addon_assets(node))
 
+    access_request = node.requests.filter(creator=auth.user).exclude(machine_state='accepted')
+    ret['user']['access_request_state'] = access_request.get().machine_state if access_request else None
```

Comment:
Good call! I'm having some trouble consolodating tho -- my first approach was in the decorators, I did try to consolodate first by moving the check to `check_contributor_auth` and raise there, but that only returns a response if it's a redirect, so couldn't pass the info along in kwargs. I also tried in `check_can_access`, but since it just returns a boolean, I don't think I can consolodate there. The other thought I had, which might be a bit extreme, would be to add a NodeRequestManager to the NodeRequest model that would always exclude accepted states by default, kind of like we do with system tags, cause as you say I dunno when we'll ever really need accepted requests at the moment, though this seems a bit extreme.  Thoughts?

Reply:
Thanks for investigating this, I think you're right, no need to go so far as to exclude accepted states by default.  I think it's fine to leave as-is.